more_itertools/more.py

```diff
@@ -1119,7 +1119,7 @@ def divide(n, iterable):
     return ret
 
 
-def always_iterable(obj):
+def always_iterable(obj, base_type=(text_type, binary_type)):
```

Comment:
I like this factoring: flexibility but not in your face. Can I propose a few other spellings that I think read better?

`always_iterable(foo, but_not=(some_type,))`
`always_iterable(foo, except_for=(some_type,))`

Reply:
Hmm, yes. I don't think either `but_not` or `except_for` really make my parameter naming dreams come true.

`ignore_types`?
`override_types`?
`no_iterate`?
`leave_alone`?
`unitary_types`?
`consider_non_iterable`?
`consider_unitary`?

 I don't love `base_type`, but we already have it for `collapse()`.

Reply:
In my mind, consistency wins, so perhaps `base_type` is best. I looked at the docs for `isinstance`, and its parameter is `class_or_tuple`, which isn't very inspiring. I do like `unitary_types`.

Reply:
I think I'm going to make the signature `always_iterable(obj, *unitary_types)` - that way it's more obvious you can pass in one argument or many.

Reply:
hmm. I don't like `*unitary_types` - that deviates from standard practice as found in `collapse()`, `isinstance()`, and in the `except` clause. Plus, it confines the signature (satisfying all positional arguments and disallowing passing by keyword):

```
>>> def always_iterable(item, *unitary_types): pass
... 
>>> always_iterable(None, unitary_types=(str,))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: always_iterable() got an unexpected keyword argument 'unitary_types'
```

Maybe the signature will never need an additional parameter, but my instinct is a single parameter is preferable.

Reply:
Good points.

Reply:
I'm still +1 for `base_type` and I would suggest only using another name for this parameter if at the same time, the library deprecates `base_type` in `collapse` in favor of the name elected for `always_iterable`.

Reply:
I didn't realize we used `base_type` elsewhere. Consistency über alles. Consider my kvetching about naming withdrawn.